### PR TITLE
use rtables pkg function for generating report

### PR DESCRIPTION
**Pull request by kartikeyakirar**
partially fixes :https://github.com/insightsengineering/teal.reporter/issues/258

- [x] using rtables pkg functions instead of creating new functions from flextable.
- [x] fixing width issue.

~~feasibility to remove any call from flextable pkg and remove it from import~~

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -31,6 +31,7 @@ Imports:
     lifecycle (>= 0.2.0),
     R6,
     rmarkdown (>= 2.19),
+    rtables (>= 0.5.1),
```
Comment (Melkiades) [inline DESCRIPTION:34]: @pawelru @shajoezhu  what do you think about moving it into imports? The library was anyway called directly in `to_flextable` (see this PR) and it was not checked (`if_installed`) beforehand, as far as I can see.

File: R/utils.R [hunk 2]
```diff
@@ -124,79 +124,35 @@ panel_item <- function(title, ..., collapsed = TRUE, input_id = NULL) {
...
   } else {
     stop(paste0("Unsupported class `(", format(class(content)), ")` when exporting table"))
   }
 
-  if (flextable::flextable_dim(ft)$widths > 10) {
-    pgwidth <- 10.5
-    width_vector <- dim(ft)$widths * pgwidth / flextable::flextable_dim(ft)$widths
-    ft <- flextable::width(ft, width = width_vector)
-  }
+  ft <- rtables::theme_docx_default(tt = content)(ft)
```
Comment (Melkiades) [inline R/utils.R:151]: If the theme is `NULL` you get something similar to the terminal print

File: DESCRIPTION (hunk 1 above)
Comment (pawelru) [reply DESCRIPTION:34]: I think it's ok. I dislike hiding dependencies in Suggests. This has a negative impact on `renv` and its dependency detection. If we cannot continue on missing package - this should go to imports.

**Comment by Melkiades** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Melkiades** [comment]
@kartikeyakirar, how is this going? I think it is kind of ready isnt it? ^^

**Comment by kartikeyakirar** [comment]
@Melkiades, I had paused work on this due to sprint priorities but have now resumed. The issue was addressed in the PR. We discussed the possibility of removing any calls to the flextable package and eliminating it from our imports. As a result, I've moved this to draft and am actively working on it.

>  feasibility to remove any call from flextable pkg and remove it from import

**Comment by kartikeyakirar** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Melkiades** [comment]
> @Melkiades, I had paused work on this due to sprint priorities but have now resumed. The issue was addressed in the PR. We discussed the possibility of removing any calls to the flextable package and eliminating it from our imports. As a result, I've moved this to draft and am actively working on it.
> 
> > Feasibility to remove any call from flextable pkg and remove it from import
> 
> I think i should have merged this and created a separate issue to remove flextable, to maintain momentum.

It is anyway imported by rtables function so I do not think it will make a lot of a difference

File: R/utils.R (hunk 2 above)
Comment (kartikeyakirar) [reply R/utils.R]: I have removed the default theme then as its not exposed for user to set.

Comment (kartikeyakirar) [reply R/utils.R]: [613f4e8](https://github.com/insightsengineering/teal.reporter/pull/265/commits/613f4e81ab77aa8880eb48d061fa8e80e316171b)

File: .github/workflows/check.yaml [hunk 3]
```diff
@@ -27,7 +27,7 @@ jobs:
     with:
       additional-env-vars: |
         _R_CHECK_CRAN_INCOMING_REMOTE_=false
-        _R_CHECK_EXAMPLE_TIMING_THRESHOLD_=10
+        _R_CHECK_EXAMPLE_TIMING_THRESHOLD_=11
```
Comment (Melkiades) [inline .github/workflows/check.yaml:30]: why is this needed?

Comment (kartikeyakirar) [reply .github/workflows/check.yaml:30]: I increased the threshold value because check failed .https://github.com/insightsengineering/teal.reporter/actions/runs/9095129970/job/24997675784.

File: R/utils.R [hunk 4]
```diff
@@ -124,76 +124,29 @@ panel_item <- function(title, ..., collapsed = TRUE, input_id = NULL) {
...
-
-    # Add footers
-    ft <- flextable::add_footer_lines(ft, flextable::as_paragraph(
-      flextable::as_chunk(mf$main_footer, props = flextable::fp_text_default(font.size = 8))
-    ))
-    if (length(mf$main_footer) > 0 && length(mf$prov_footer) > 0) ft <- flextable::add_footer_lines(ft, c("\n"))
-    ft <- flextable::add_footer_lines(ft, flextable::as_paragraph(
-      flextable::as_chunk(mf$prov_footer, props = flextable::fp_text_default(font.size = 8))
-    ))
+    ft <- rtables::df_to_tt(df)
```
Comment (Melkiades) [inline R/utils.R:135]: it can make sense, I think we lose some info though

Comment (Melkiades) [reply R/utils.R:135]: but from my tests not enough so good to go this ;)

File: R/utils.R [hunk 5]
```diff
@@ -124,76 +124,29 @@ panel_item <- function(title, ..., collapsed = TRUE, input_id = NULL) {
 #'
 #' @keywords internal
 to_flextable <- function(content) {
-  if (inherits(content, c("rtables", "TableTree", "ElementaryTable", "listing_df"))) {
+  if (inherits(content, c("rtables", "TableTree", "ElementaryTable"))) {
+    ft <- rtables::tt_to_flextable(content)
+  } else if (inherits(content, "listing_df")) {
     mf <- rtables::matrix_form(content)
```
[the remark this answers is not in the document]
Comment (kartikeyakirar) [reply R/utils.R:130]: Interesting. The result is exactly the same when I checked it for some modules, e.g. tm_g_pp_adverse_events.
image

[the remark this answers is not in the document]
Comment (kartikeyakirar) [reply R/utils.R]: but make sense to keep `rlistings::matrix_form `[2a68bf2](https://github.com/insightsengineering/teal.reporter/pull/265/commits/2a68bf233fb2993005cf59064b251490c1b8b9c7)

**Review by Melkiades — APPROVED** [review]
It looks good to me. Only could you reinstate some tests for the functionality anyway? something simple ofc. Thank you very much Kartik!!

File: R/utils.R (hunk 5 above)
[the remark this answers is not in the document]
Comment (Melkiades) [reply R/utils.R]: I worked recently on the two matrix_form functions and have some key diffs but are mainly related to pagination and exports so here was still fine. Still better to use the correct function ;)

**Comment by kartikeyakirar** [comment]
> some tests for the functionality anyway? 

done [800d071](https://github.com/insightsengineering/teal.reporter/pull/265/commits/800d071a493311adbeff15c514db5d13584b0915)

File: R/utils.R [hunk 6]
```diff
@@ -124,76 +124,31 @@ panel_item <- function(title, ..., collapsed = TRUE, input_id = NULL) {
...
-    ))
+    ft <- rtables::df_to_tt(df)
+    if (length(mf$main_title) != 0) {
+      rtables::main_title(ft) <- mf$main_title
+    }
+    rtables::subtitles(ft) <- mf$subtitles
+    rtables::main_footer(ft) <- mf$main_footer
+    rtables::prov_footer(ft) <- mf$prov_footer
+    rtables::header_section_div(ft) <- mf$header_section_div
+    ft <- rtables::tt_to_flextable(ft, total_width = c(grDevices::pdf.options()$width - 1))
```
[the remark this answers is not in the document]
Comment (kartikeyakirar) [reply R/utils.R:143]: the size is too large, and any width adjustment cannot handle it. Please check the table https://github.com/insightsengineering/teal.reporter/issues/274#issuecomment-2137175218. We need to break the table with similar functions as `export_as_pdf`.
